GoogleForCreators/web-stories-wp
# Add filter for all media

## Context



## Summary
Previously, filter was allowed only for background media. This PR adds filtering for video and media elements, too.


## Relevant Technical Choices
- Migrates `backgroundOverlay` prop to `overlay` for making combining elements easier.


## To-do



## User-facing changes
Filtering is now allowed for media, too.


## Testing Instructions
I Migration test
BEFORE checking out this PR, create a new story.
1. Add 3 pages with background media, all with different filters (Tint, Linear, Vignette)
2. Now check out this PR
3. Verify that all the pages still have the correct Filter as applied before.

II Filter test
1. Add a media (test with both, image and video)
2. Test the Filter panel by applying different filters and verifying it looks as expected both in the editor and in preview
3. Verify that in case of mixed multi-selection values only the colour field is displayed.

III Merging / drop-targeting
1. Add background media with filter
2. Add another media with filter.
3. Drag the media into the background. 
4. Verify that the filter from the dragged media is applied

IV Merging / drop-targeting
1. Add two media elements, one with a filter.
2. Now drag the media with filter into the media without filter.
3. Verify the combination does not have a filter.


### QA





This PR can be tested by following these steps:

1. See the instructions above.

### UAT







This PR can be tested by following these steps:

1.
## Checklist




---



Fixes #6526
